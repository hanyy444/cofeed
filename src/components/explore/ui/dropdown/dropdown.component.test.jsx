import { render, screen } from "@testing-library/react";
import { vi } from "vitest";
import Dropdown from "./dropdown.component";

describe('Test dropdown component', () => { 
    it('Should render without crashing', () => {
        expect.assertions(3)
        render(<Dropdown 
            showDropDown={true} 
            toggleDropdown={()=>vi.fn()}
            navItems={[{ text: 'test', handler: vi.fn() }]}/>)
        const dropdown = screen.getByTestId('dropdown')
        expect(dropdown).toBeInTheDocument()
        expect(dropdown).toMatchSnapshot()
        expect(dropdown).toHaveTextContent('test')
    })
 })