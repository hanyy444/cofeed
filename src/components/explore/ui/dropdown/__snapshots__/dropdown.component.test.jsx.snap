// Vitest Snapshot v1

exports[`Test dropdown component > Should render without crashing 1`] = `
<div
  class="dropdown"
  data-testid="dropdown"
>
  <img
    alt=""
    class="dropdown__img"
    src="http://localhost:3000/public/assets/undefined"
  />
  <ul
    class="dropdown__list"
  >
    <li>
      test
    </li>
  </ul>
</div>
`;
